components/shared/embedder/lib.rs

```diff
@@ -921,3 +928,10 @@ pub enum JavaScriptEvaluationError {
     /// value into a [`JSValue`].
     SerializationError,
 }
+
+#[derive(Clone, Copy, Debug, Deserialize, Serialize)]
+pub struct RgbColor {
+    pub red: u8,
+    pub green: u8,
+    pub blue: u8,
```

Comment:
I'm not sure if you've see this: https://github.com/whatwg/html/issues/3400. Essentially this might have to change in the future, or we could even reuse data types from the [color](https://docs.rs/color/0.3.0/color/) crate. This is probably okay for now though!

Reply:
I didn't know about this proposal. Interesting!

We could make the color an opaque type that only exposes getter functions for rgb components and something like `from_rgb(..) -> Self;`. That way we can implement color spaces in a followup without breaking changes to the embedding api.

Reply:
I think that we will likely go over the entire API at some point before cutting a point release. We could consider it then.